elk/tests/test_extraction_output.py

```diff
@@ -1,17 +1,66 @@
 import pytest
+import os
 import pandas as pd
+import pickle
+from pathlib import Path
 
+# import train and evaluate
+from elk.train import train
+from elk.evaluate import evaluate
+import json
 
-class TestExtractionOutput:
-    def testExtractionOutput(self):
-        # Read the contents of the csv files into pandas dataframes
-        # TODO: Right now, the file paths are hardcoded. Improve on that.
-        correct_output_file = pd.read_csv(
-            "./test_data/deberta-v2-xxlarge-mnli_confusion_0.csv"
-        )
-        current_output_file = pd.read_csv(
-            "../evaluation_results/deberta-v2-xxlarge-mnli_confusion_0.csv"
-        )
+default_config_path = Path(__file__).parent.parent / "default_config.json"
 
-        # Compare the contents of the dataframes
-        assert correct_output_file.equals(current_output_file)
+with open(default_config_path, "r") as f:
+    default_config = json.load(f)
+datasets = default_config["datasets"]
+models = default_config["models"]
+prefix = default_config["prefix"]
+models_layer_num = default_config["models_layer_num"]
+
+
+def train_and_evaluate(model_name, dataset):
+    """
+    Train and evaluate the model with the given dataset.
+    """
+    training_command = "python -m elk.train "
+    f"--model {model_name} --prefix normal --dataset {dataset} --num_data 1000 --seed 0"
+    evaluation_command = "python -m elk.evaluate "
+    f"--model {model_name} --prefix normal --dataset {dataset} --num_data 1000 --seed 0"
+    os.system(training_command)
+    os.system(evaluation_command)
+
+
+def test_model_performance(model_name, dataset):
+
+    """
+    TODO: implement evaluation of the model with all datasets
+    # in progress
+    results_path = Path(__file__).parent.parent / f"{model_name}_confusion_0.csv"
+
+    for ccs and logistic regression
+
+    generate result different method
+    """
+
+    # Train and evaluate the model
+    train_and_evaluate(model_name, dataset)
+
+    # Read the contents of the csv files into pandas dataframes
+    benchmark_performance = pd.read_csv(f"../test_data/{model_name}_confusion_0.csv")
```

Comment:
You might want to create this csv file (probably manually) and add it to the repo. It should contain all the different combinations from the table in the paper (for logistic regression and CCS only): https://ar5iv.labs.arxiv.org/html/2212.03827/assets/x8.png